File: python/test_server.py
import time
import zmq
import struct
import common_pb2 as common

context = zmq.Context()
socket = context.socket(zmq.REP)
socket.bind("tcp://*:5555")

ping = common.PingServer().SerializeToString()

while True:
  message = socket.recv()
  socket.send(ping)

File: test_server.cpp
#include <string>
#include <zmq.hpp>

int main()
{
  printf("Hello World!\n");
   zmq::context_t context(1);
   zmq::socket_t sock(context, zmq::socket_type::req);
   sock.bind("tcp://*:5555");
  //  while (true) {
  //   //  zmq::message
  //  }
   const std::string_view m = "Hello, world";
   sock.send(zmq::buffer(m), zmq::send_flags::dontwait);
  printf("Hello World!\n");
}